mlflow/openai/utils/chat_schema.py

```diff
@@ -285,6 +285,14 @@ def _find_tool_message(messages, tool_type):
 def _parse_tools(inputs: dict[str, Any]) -> list[ChatTool]:
     tools = inputs.get("tools", [])
 
+    try:
```

Comment:
Can we just check if tools is an iterable? We should avoid importing non-public classes from external libraries